syntax = "proto3";

message PersonMessage {
  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string company_name = 4;
}

message PersonMessageList {
  repeated PersonMessage persons = 1;
}

message GetPersonMessage {
  repeated int32 id_list = 1;
}

message EmptyPersonMessage {}


service PersonService {
  rpc Create(PersonMessage) returns (PersonMessage);
  rpc Get(GetPersonMessage) returns (PersonMessageList);
  rpc GetAll(EmptyPersonMessage) returns (PersonMessageList);
}